src/lib.rs

```diff
@@ -29,13 +32,20 @@ macro_rules! regex {
 const CAPTURE_NAME_FOR_TREE_SITTER_GREP: &str = "_tree_sitter_lint_capture";
 const CAPTURE_NAME_FOR_TREE_SITTER_GREP_WITH_LEADING_AT: &str = "@_tree_sitter_lint_capture";
 
-pub fn run(_args: Args) {
+pub fn run(args: Args) {
     let language = tree_sitter_rust::language();
     let context = Context::new(language);
     let resolved_rules = get_rules()
         .into_iter()
+        .filter(|rule| match args.rule.as_ref() {
+            Some(rule_arg) => &rule.name == rule_arg,
+            None => true,
+        })
         .map(|rule| rule.resolve(&context))
         .collect::<Vec<_>>();
+    if resolved_rules.is_empty() {
```

Comment:
This makes sense, but took me a minute to think through how using a filter in the "single rule provided as an argument" case makes sense (because I expected a `find` or something).

Reply:
No I know this is gross, once there's "dynamic configuration of which rules you have enabled" then we have to check separately whether "there were no rules enabled in the first place (before attempting to narrow via a `--rule` command-line argument)"

Maybe that's technically kind of orthogonal in that then if `--rule` is present we could still do something based around `.find()` to "narrow" it